x-pack/legacy/plugins/code/public/components/integrations/code_viewer.tsx

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { Component } from 'react';
+import { editor } from 'monaco-editor';
+import { ResizeChecker } from 'ui/resize_checker';
+import {
+  EuiFlexGroup,
+  EuiProgress,
+  EuiPage,
+  EuiPageBody,
+  EuiPageContent,
+  EuiPageContentHeader,
+  EuiPageContentHeaderSection,
+  EuiPageContentBody,
+} from '@elastic/eui';
+import { monaco } from '../../monaco/monaco';
+import { requestFile } from '../../sagas/file';
+import { GitBlame } from '../../../common/git_blame';
+import { BlameWidget } from '../../monaco/blame/blame_widget';
+import { requestBlame } from '../../sagas/blame';
+
+export interface Props {
+  repo: string;
+  file: string;
+  revision: string;
+  showBlame?: boolean;
+}
+
+interface State {
+  loading: boolean;
+}
+
+export class CodeViewer extends Component<Props, State> {
+  private ed?: editor.IStandaloneCodeEditor;
+  public blameWidgets: any;
+  private lineDecorations: string[] | null = null;
+  private resizeChecker?: ResizeChecker;
+  private viewerRef: React.RefObject<HTMLDivElement>;
+
+  constructor(props: Props, context: any) {
+    super(props, context);
+    this.state = {
+      loading: true,
+    };
+    this.viewerRef = React.createRef();
```

Comment:
You don't strictly need a constructor here, and you could make both `state` and `viewerRef` instance members instead: [example for createRef](https://github.com/elastic/kibana/blob/master/x-pack/legacy/plugins/code/public/components/code_block/code_block.tsx#L41).